File: data/src/main/java/home/oleg/placesnearme/mapper/LocationMapper.java
package home.oleg.placesnearme.mapper;

import android.location.Location;

import home.oleg.placenearme.models.UserLocation;

/**
 * Created by Oleg Sheliakin on 17.09.2018.
 * Contact me by email - olegsheliakin@gmail.com
 */
public final class LocationMapper {

    private LocationMapper() {

    }

    public static UserLocation map(Location location) {
        return new UserLocation(location.getLatitude(), location.getLongitude());
    }

    public static Location map(UserLocation userLocation) {
        Location location = new Location("");
        location.setLatitude(userLocation.getLat());
        location.setLongitude(userLocation.getLng());
        return location;
    }

    public static home.oleg.placenearme.models.Location map(home.oleg.placesnearme.network.models.Location location) {
        home.oleg.placenearme.models.Location result = new home.oleg.placenearme.models.Location();
        result.setAddress(location.getAddress());
        result.setCc(location.getCc());
        result.setCity(location.getCity());
        result.setCountry(location.getCountry());
        result.setCrossStreet(location.getCrossStreet());
        result.setFormattedAddress(location.getFormattedAddress());
        result.setLat(location.getLat());
        result.setLng(location.getLng());
        return result;
    }

}

File: core-data/src/main/java/home/oleg/placesnearme/coredata/mapper/LocationMapper.kt
package home.oleg.placesnearme.coredata.mapper

import android.location.Location
import home.oleg.placesnearme.coredomain.models.LatLng

import home.oleg.placesnearme.coredomain.models.UserLocation
import home.oleg.placesnearme.corenetwork.models.LocationDto

/**
 * Created by Oleg Sheliakin on 17.09.2018.
 * Contact me by email - olegsheliakin@gmail.com
 */
object LocationMapper {

    fun map(location: Location): LatLng {
        return LatLng(location.latitude, location.longitude)
    }

    fun map(userLocation: UserLocation): Location {
        val location = Location("")
        location.latitude = userLocation.lat
        location.longitude = userLocation.lng
        return location
    }

    fun map(location: LocationDto) = home.oleg.placesnearme.coredomain.models.UserLocation(
            address = location.formattedAddress.firstOrNull(),
            lat = location.lat,
            lng = location.lng
    )

}
